### PR TITLE
feat(filter-box): hide druid options if druid not enabled

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
Currently the filter box control panel shows legacy Druid specific controls even when legacy Druid isn't enabled. This PR removes the Druid specific sections if legacy Druid is not enabled, and also moves the controls to their own rows. In addition, a new property is added to `bootstrapData` to list installed optional Python dependencies to be able to customize control panels that depend on this functionality (in this case this will be used to show/hide the forecast section in the ECharts timeseries plugin).

### BEFORE
Currently the SQL and Druid sections labels overlap each other:
image

### AFTER
With `DRUID_IS_ACTIVE` set to true:
image

With `DRUID_IS_ACTIVE` set to false:
image

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/visualizations/FilterBox/controlPanel.jsx [hunk 1]
```diff
@@ -20,6 +20,37 @@ import React from 'react';
 import { t } from '@superset-ui/core';
 import { sections } from '@superset-ui/chart-controls';
 
+const appContainer = document.getElementById('app');
+const bootstrapData = JSON.parse(appContainer.getAttribute('data-bootstrap'));
```
Comment (kgabryje) [inline superset-frontend/src/visualizations/FilterBox/controlPanel.jsx:24]: I think that we usually parse bootstrapped data in the main index file of a given view. Do you think we could move those 2 lines higher in the files hierarchy as well?

Comment (michael-s-molina) [reply superset-frontend/src/visualizations/FilterBox/controlPanel.jsx:24]: In `src/explore/index.jsx`

```
const exploreViewContainer = document.getElementById('app');
const bootstrapData = JSON.parse(
  exploreViewContainer.getAttribute('data-bootstrap'),
);
initFeatureFlags(bootstrapData.common.feature_flags);
```

**Review by nytai — APPROVED** [review]

**Comment by yousoph** [comment]
🏷️ 2021.21

**Review by rusackas — APPROVED** [review]
LGTM, and manual tests/QA looks good after the last commit.

**Pull request merged**